### PR TITLE
Generate '@Deprecated' Annotation for Deprecated Proto Fields

### PR DESCRIPTION
**Pull request by nichite**
This change brings Dart closer in line to other languages like Java, which already generate this annotation. It should be dropped in dart2js and have no real code impact size for production users.

### CODE REVIEW

File: protoc_plugin/lib/file_generator.dart [hunk 1]
```diff
@@ -292,7 +292,7 @@ class FileGenerator extends ProtobufContainer {
 
     out.println('// ignore: UNUSED_SHOWN_NAME');
     out.println(
-        '$_coreImport show int, bool, double, String, List, Map, override;\n');
+        '$_coreImport show int, bool, double, String, List, Map, override, Deprecated;\n');
```
Comment (sigurdm) [inline protoc_plugin/lib/file_generator.dart]: Now you're here, could you sort this show-list?

**Comment by sigurdm** [comment]
Do you plan to also support deprecated messages, enums, files etc.?

File: protoc_plugin/lib/file_generator.dart (hunk 1 above)
Comment (nichite) [reply protoc_plugin/lib/file_generator.dart]: Done.

**Comment by nichite** [comment]
I think adding support for the messages, etc is probably a good idea too. I'll look into what the other languages are doing.

**Pull request merged**